### PR TITLE
New Blog Post - Elasticsearch Aggregations

### PR DESCRIPTION
**Pull request by tonysm**

### CODE REVIEW

**Comment by tonysm** [comment]
\cc @Anahkiasen @dieterve @hannesvdvreken could you guys review?

File: _posts/2015-02-08-elasticsearch-aggregations.md [hunk 1]
```diff
@@ -0,0 +1,469 @@
+---
+layout: post
+title: "Elasticsearch Aggregations"
+date:   2015-02-09
+author: tony
+categories: [elasticsearch, aggregations,search]
+---
+
+I have already covered how you can easily [integrate Elasticsearch with your app](http://blog.madewithlove.be/post/integrating-elasticsearch-with-your-laravel-app/), but I haven't talked anything about how you can *query* your data, and I believe this is not covered (I guess it's because it depends on the context you are trying to perform your query and what you really need).
```
Comment (Anahkiasen) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: Not covered by what?

File: _posts/2015-02-08-elasticsearch-aggregations.md [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+<!-- more -->
+
+I won't cover the basics of querying or filtering here, instead I will cover a really cool feature called *aggregations*, it's a really way to perform some analysis over your data. And I'm also going to cover a still *experimental* feature called [**scripted metric**](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-aggregations-metrics-scripted-metric-aggregation.html).
+
+To start, you need to setup Elasticsearch locally or in a VM (see here MISSING THE LINK). You will also need some schema and data. You can use the example given in the documentation:
```
Comment (Anahkiasen) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: "Missing the link" :d

Comment (tonysm) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: haha, nice catch

File: _posts/2015-02-08-elasticsearch-aggregations.md [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+* `reduce_script`: the combine_script trasnformed our aggregation value into an array of `amounts`, we can *reduce*  it to a single value in this script (again, those familiar with `map/reduce` concept already knew this);
+
+You can try performing this aggregation removing each script (except for the `map_script` that is the only one required and for the `init_script`that our `map_script` uses). It clarifies a little more. :)
+
+### What is available for the `map_script` stage?
+
+Well, pretty much you have `doc`, which is (of course) your document itself. But you also have a `_source` object, which corresponds to the source of your document (this one is slower than the `doc` object).
+
+The down side of using scripts in aggregations is that we can end up with a bunch of Java (Groovy) code that we have to take care. You can store your scripts your scripts in elasticsearch and just reference them, but you need to be careful with it and treat it well.
```
Comment (hannesvdvreken) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: 2 times "your scripts" right after each other.

Comment (hannesvdvreken) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: "code that we have to take care.". I think you were aiming for "code that we have to take care **of**.", no?

File: _posts/2015-02-08-elasticsearch-aggregations.md [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+                    "score" : { "type" : "integer" }
+                }
+            }
+        }
+    }
+}
+
+{% endhighlight %}
+
+This mapping let you have a `matches` document with nested `players`, you can have an array of players in a single match. Let's say you want to display the "top players" and it is just a SUM of all scores of each player, then you could use nested aggregations like so:
```
Comment (hannesvdvreken) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: "this mapping let you", third person singular, so: "this mapping let**s** you"

File: _posts/2015-02-08-elasticsearch-aggregations.md [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+            "avg": 7.571428571428571,
+            "sum": 106
+         }
+      }
+   }
+}
+
+{% endhighlight %}
+
+Pretty cool. The examples above shows us how you can have nested documents and how you can nest/chain aggregations (not related to nested documents).
```
Comment (hannesvdvreken) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: "The examples above shows us" -> third person plural, so "show" without "s", or use "the example above" instead ;-)

File: _posts/2015-02-08-elasticsearch-aggregations.md [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+
+You can read more about the [nested type](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/mapping-nested-type.html#_mapping) and [nested aggregations](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-aggregations-bucket-nested-aggregation.html) in the docs.
```
Comment (hannesvdvreken) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: I'd put the link in [in the docs]() instead of on [nested type]() but that's just my opinion.

**Comment by hannesvdvreken** [comment]
Very nice blog post!

I'd like to see a comparison blog post on MapReduce performance across document stores CouchDB, MongoDB and ES. I'll put that on my to do list.

File: _posts/2015-02-08-elasticsearch-aggregations.md (hunk 6 above)
Comment (tonysm) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: Yeah, I thought about it, but there are two links there. =/

Comment (hannesvdvreken) [inline _posts/2015-02-08-elasticsearch-aggregations.md]: oh i see now

**Comment by hannesvdvreken** [comment]
:ship:

**Comment by hannesvdvreken** [comment]
When do you want to publish this?

**Comment by tonysm** [comment]
Well, when ever you guys want. :)

**Comment by andreascreten** [comment]
Friday?

**Comment by hannesvdvreken** [comment]
@github when can we start to shedule PR merges?

**Comment by hannesvdvreken** [comment]
Shall we deploy this?

**Comment by dieterve** [comment]
:shipit:

**Comment by tonysm** [comment]
:ship:

**Comment by tonysm** [comment]
Should we change the date?

**Comment by hannesvdvreken** [comment]
yeah

**Comment by hannesvdvreken** [comment]
Wait, why is my gitattributes blog post in there? Did I merge that in there?

**Comment by tonysm** [comment]
uhmm, looks like you did. haha

**Comment by hannesvdvreken** [comment]
Fixed it.

**Pull request merged**